### PR TITLE
Add support for cmt (Chromium Multitouch drivers), custom drivers for chromebooks.

### PR DESCRIPTION
**Pull request by kcalvinalvin**
###### Motivation for this change
Add support for custom Chromebook drivers.

###### Things done
Added following cmt dependencies to pkgs: libgestures, libevdevc, and chromium-xorg-conf to pkgs
Added cmt.nix to nixpkgs/nixos/modules/servers/x11/hardware/

Known bug: Can't adjust acceleration or sensitivity from system touchpad settings (tested on samus).

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @yegortimoshenko

### CODE REVIEW

File: nixos/modules/services/x11/hardware/cmt.nix [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+         type = types.bool;
+         default = false;
+         description = "Enable chrome multitouch input (cmt). Touchpad drivers that are configured for chromebooks";
+      };
+      atlas = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable custom touchpad for selected model";
+      };
+      banjo = mkOption {
```
Comment (Infinisil) [inline nixos/modules/services/x11/hardware/cmt.nix]: This is not good, there shouldn't be a separate option for every possible model. Instead I suggest to have a single option like this:
```
{
  models = mkOption {
    type = types.enum [ "atlas" "banjo" ... ];
    default = [];
    example = [ "banjo" ];
    description = "Which models to enable cmt for.";
  };
}
```

Or maybe even not have an option at all and always enable all of them, I don't see any disadvantage of doing that

File: nixos/modules/services/x11/hardware/cmt.nix [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+
+in {
+
+  options = {
+
+    services.xserver.cmt = { 
+      enable = mkOption {
+         type = types.bool;
+         default = false;
+         description = "Enable chrome multitouch input (cmt). Touchpad drivers that are configured for chromebooks";
```
Comment (Infinisil) [inline nixos/modules/services/x11/hardware/cmt.nix]: Put a `.` at the end of descriptions

File: pkgs/os-specific/linux/xf86-input-cmt/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  ];
+
+  configureFlags = [
+    "--with-sdkdir=${placeholder "out"}"
+  ];
+
+  meta = {
+    description = "Chromebook touchpad driver";
+    license = stdenv.lib.licenses.mit;
+    platforms = stdenv.lib.platforms.linux;
```
Comment (Infinisil) [inline pkgs/os-specific/linux/xf86-input-cmt/default.nix]: Homepage and maintainer fields are missing

File: pkgs/os-specific/linux/libgestures/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+      --replace -Werror -Wno-error \
+      --replace '$(DESTDIR)/usr/include' '$(DESTDIR)/include'
+  '';
+
+  nativeBuildInputs = [ pkgconfig ];
+  buildInputs = [ glib jsoncpp ];
+
+
+  makeFlags = [ "DESTDIR=$(out)" "LIBDIR=/lib" ];
+}
```
Comment (Infinisil) [inline pkgs/os-specific/linux/libgestures/default.nix]: `meta` missing completely, in `libevdevc` as well

File: pkgs/os-specific/linux/xf86-input-cmt/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, fetchFromGitHub, pkgconfig, xorgserver, xorgproto,
+  utilmacros, libgestures, libevdevc, chromium-xorg-conf }:
```
Comment (Infinisil) [inline pkgs/os-specific/linux/xf86-input-cmt/default.nix]: `chromium-xorg-conf` is never used

**Review by infinisil — CHANGES_REQUESTED** [review]
Left some comments. You should also fix the formatting and eventually get rid of the merge conflicts and have a clean git history

**Pull request merged**